nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-site-to-site/src/main/java/org/apache/nifi/remote/PeerDescriptionModifier.java

```diff
@@ -0,0 +1,160 @@
+            final String targetHostName = hostname.evaluateExpressions(variables, null);
+            if (isBlank(targetHostName)) {
+                throw new IllegalStateException("Target hostname was not resolved for the route definition " + name);
+            }
+
+            final String targetPortStr = port.evaluateExpressions(variables, null);
+            if (isBlank(targetPortStr)) {
+                throw new IllegalStateException("Target port was not resolved for the route definition " + name);
+            }
+
+            String targetIsSecure = secure == null ? null : secure.evaluateExpressions(variables, null);
+            if (isBlank(targetIsSecure)) {
+                targetIsSecure = "false";
+            }
+            return new PeerDescription(targetHostName, Integer.valueOf(targetPortStr), Boolean.valueOf(targetIsSecure));
+        }
+    }
+
+    private Map<SiteToSiteTransportProtocol, List<Route>> routes;
+
+
+    private static final String PROPERTY_PREFIX = "nifi.remote.route.";
+
+    public PeerDescriptionModifier(final NiFiProperties properties) {
+        final Map<String, List<String>> routeDefinitions = properties.getPropertyKeys().stream()
+                .filter(k -> k.startsWith(PROPERTY_PREFIX))
+                .collect(Collectors.groupingBy(k -> k.substring(PROPERTY_PREFIX.length(), k.lastIndexOf('.'))));
+
+        routes = routeDefinitions.entrySet().stream().map(r -> {
+            final Route route = new Route();
+            final String[] key = r.getKey().split("\\.");
+            route.protocol = SiteToSiteTransportProtocol.valueOf(key[0].toUpperCase());
+            route.name = key[1];
+            r.getValue().forEach(k -> {
+                final String name = k.substring(k.lastIndexOf('.') + 1);
+                final String value = properties.getProperty(k);
+                switch (name) {
+                    case "when":
+                        route.predicate = Query.prepare(value);
+                        break;
+                    case "hostname":
+                        route.hostname = Query.prepare(value);
+                        break;
+                    case "port":
+                        route.port = Query.prepare(value);
+                        break;
+                    case "secure":
+                        route.secure = Query.prepare(value);
+                        break;
+                }
+            });
+            return route;
+        }).filter(Route::isValid).collect(Collectors.groupingBy(r -> r.protocol));
+
+    }
+
+    private void addVariables(Map<String, String> map, String prefix, PeerDescription peer) {
+        map.put(format("%s.hostname", prefix), peer.getHostname());
+        map.put(format("%s.port", prefix), String.valueOf(peer.getPort()));
+        map.put(format("%s.secure", prefix), String.valueOf(peer.isSecure()));
+    }
+
+    public boolean isModificationNeeded(final SiteToSiteTransportProtocol protocol) {
```

Comment:
@mcgilman The reason of this method being separated is to avoid unnecessary object creations at the caller. I imagine most of deployments will not use this configuration, hence if caller side (e.g. SiteToSiteResource) always create objects to call modify unconditionally, but doing nothing actually because not having modification definitions, that would be waste  or resources.